static/js/ui_report.ts

```diff
@@ -59,23 +73,23 @@ export function generic_embed_error(error_html) {
     );
 }
 
-export function generic_row_button_error(xhr, btn) {
-    if (xhr.status.toString().charAt(0) === "4") {
+export function generic_row_button_error(xhr: JQuery.jqXHR, btn: JQuery): void {
+    if (xhr.status.toString().startsWith("4")) {
         btn.closest("td").html(
-            $("<p>").addClass("text-error").text(JSON.parse(xhr.responseText).msg),
+            $("<p>").addClass("text-error").text(JSON.parse(xhr.responseText).msg)[0],
         );
```

Comment:
It seems wrong to pass an `HTMLElement` to `.html()`—why does TypeScript allow this?

I think this was supposed to be `btn.closest("td").empty().append($(…).addClass(…).text(…))`.

Reply:
I think the jQuery types are correct there (unfortunately) -- passing `HTMLElement` and even a `JQuery` object that is not in types work as expected when I tested both. What I didn't know was that this isn't a documented feature. I updated this in a new prep commit.